lib/buildkite/test_collector/ci.rb

```diff
@@ -68,7 +70,8 @@ def github_actions
       "url" => File.join("https://github.com", ENV["GITHUB_REPOSITORY"], "actions/runs", ENV["GITHUB_RUN_ID"]),
       "branch" => ENV["GITHUB_REF_NAME"],
       "commit_sha" => ENV["GITHUB_SHA"],
-      "number" => ENV["GITHUB_RUN_NUMBER"],
+      "build_number" => ENV["GITHUB_RUN_NUMBER"],
+      "build_id" => ENV["GITHUB_RUN_ID"]
```

Comment:
According to the [GitHub docs](https://docs.github.com/en/actions/learn-github-actions/variables), `GITHUB_RUN_ID` doesn't change upon re-run. I'm not sure this is suitable for our purposes. We can combine `GITHUB_RUN_ATTEMPT` with `GITHUB_RUN_ID` to get a unique value upon re-run.

We could also used the value created on line 69. It uses different env but achieves the same result.